init-interfaces.sh

```diff
@@ -42,16 +42,18 @@ set_description() {
   local mac=$1
   local nic=$2
   local description=$3
-  local cons=$(egrep -l -i "mac-address=$mac|interface-name=$nic" $(egrep -l "type=ethernet" /etc/NetworkManager/system-connections/*))
-  for c in $cons; do
-      if ! grep nmstate.interface.description $c; then
-         echo "" >> $c
-         echo "[user]" >> $c
-         echo nmstate.interface.description=$description >> $c
+  local connections=$(grep -REl "type=ethernet" /etc/NetworkManager/system-connections | xargs -I{} -- grep -El -i "mac-address=${mac}|interface-name=${nic}" "{}")
+  IFS=$'\n'
```

Comment:
please keep all the changed related to space names in this commit, and move all other semantic changes (like `$c` --> `"${connection}"`) to a different commit. This other commit should make it clear that it doesn't change the script logic.